### DISCUSSION TITLE
v2.10.0

### DISCUSSION THREAD

**Discussion by mikecao** [umami-software/umami#2536] [General]
# Features
## Improved teams collaboration

This release we're introducing a new way of working with teams. Now instead of simply sharing websites through a team, all team members can create, edit, view and delete websites in a team.

Additionally, teams are now clearly separate from your personal account. You will always be viewing the application in the context of your personal account or your team. That means pages like the Dashboard, Websites, and Reports now only show you the relevant websites.

A new dropdown has been added to navigation bar which lets you quickly switch between your teams and personal account:

![image](https://github.com/umami-software/umami/assets/477975/2ef15212-58f7-4551-be68-a811dea04214)

See this [blog post](https://umami.is/blog/improved-teams-collaboration?ref=github-release) for more details about the new teams feature.

## Searchable dropdowns

A new UI element we've added across the application are searchable dropdowns. This is especially useful for users with a lot of websites.

![image](https://github.com/umami-software/umami/assets/477975/2b8dfc9a-0a8d-4233-8545-0928374d56e8)

## New Funnel report display

The Funnel report now has a new display component that makes it easier to view your funnel data. Each step is now clearly shown with statistics showing how much of the audience has dropped off at eas

![image](https://github.com/umami-software/umami/assets/477975/d5e0307e-3817-4194-82c6-4dfada81e24a)

## Improved tracker

Several important changes were made to the tracker script:

- Title, URL, and referrer are not URI encode to support Unicode characters for languages like Chinese.
- Improved the handling of elements like anchor tags and buttons tags when using custom events.
- Removed the do-not-track code as the feature was not widely supported.

# Fixes
- Fixed missing edit buttons #2422
- Fixed issue with view-only role #2445 
- Fixed issue with click events not triggering #2471
- Fixed issue with handing Chinese characters #2470 
- Fixed issue with metrics search #2505 

# Updates
- Added OCI labels for Docker #2512 
- Updated selection for ES language #2438 
- Language updates: Swedish, Mongolian, Arabic
- Upgraded Next `v14.1.0`
- Upgraded Prisma `v5.9.1`

A huge thanks to all the contributors on this release! @akshar-dave @Braggedtooth @jaasonw @spencertweedy @timokoessler @enfiniteio @bilguun0203 @Maxime-J @jermanuts @pxlfrk 

<hr /><em>This discussion was created from the release <a href='https://github.com/umami-software/umami/releases/tag/v2.10.0'>v2.10.0</a>.</em>

**Comment by mobeicanyue** [comment]
`do-not-track` is useful for me, remove it is a pity.

**Reply by mikecao** [reply]
You can still implement it yourself pretty easily. Here is the code:

```javascript
const doNotTrack = () => {
  const { doNotTrack, navigator, external } = window;

  const msTrackProtection = 'msTrackingProtectionEnabled';
  const msTracking = () => {
    return external && msTrackProtection in external && external[msTrackProtection]();
  };

  const dnt = doNotTrack || navigator.doNotTrack || navigator.msDoNotTrack || msTracking();

  return dnt == '1' || dnt === 'yes';
};
```

Then just show/hide the tracking script accordingly.

**Reply by mobeicanyue** [reply]
It is front code or backend?

**Reply by mikecao** [reply]
Front end, you have to read the value from the user's browser.

**Reply by pointout** [reply]
Thanks for the code, but in my opinion "do-not-track" should be respected by default with an opt-out parameter in the track script.

**Reply by mobeicanyue** [reply]
I agree.

**Reply by mobeicanyue** [reply]
And implement `do-not-track` need to write js file and mount it to website, which may be block.
But it will be very convenient if umami implement itself.
Sincerely request you to reconsider it, please.

**Comment by Matthijz98** [comment]
is the migrate website to teams function also coming to self hosted?

**Reply by mikecao** [reply]
Yes, just released https://github.com/umami-software/umami/releases/tag/v2.10.1

**Comment by philippgitpush** [comment]
Great work! I have one little bit of feedback on this: As an "admin" user, it is now no longer possible to have a dashbaord view where all sites (that you technically have access to) are listed in one place. Since sites are now transferred between teams, you have to switch to a team's view for each site listed in a team.

I used to use my administrator account to view all sites and share individual access via teams with different users.
I'm not sure if it was intentional to roll this out to everyone, even the administrator. It's basically just a change of habits, but as an administrator I miss having all sites listed in my dashboard. 

**To illustrate this in more detail:**

I have listed 5 websites. As an administrator, I can see all 5 websites on my dashboard. I have 4 other users who each have a website (the owners of the website, so to speak) who only have access to their website via teams. As an administrator I was still able to list all websites on the dashboard. Since the update, I now have to click on the respective team with a few more clicks and then view the dashboard there. Repeat this process for X teams and the time will stack up.

**Reply by mikecao** [reply]
We will be adding an admin context in a future release. We already have user and team context, so the admin will be able to see all websites, build dashboards from all websites, etc.
